### DISCUSSION TITLE
为什么milvus1.x向量数据迁移到2.3里，使用相同的向量数据搜索出的结果不一样？

### DISCUSSION THREAD

**Discussion by zxmin205** [milvus-io/milvus#29441] [Q&A and General discussion]
从milvus1.x向量数据迁移到2.3里，collection和index的创建的参数都一样（INDEX_TYPE = 'IVF_SQ8' ，METRIC_TYPE = 'L2'），使用相同的向量数据进搜索，TOP10出来的数据不一样，相似度1.x第一条是对的
![微信截图_20231224215358](https://github.com/milvus-io/milvus/assets/21117028/b11f8016-5678-453e-807a-0935bd212795)
![微信截图_20231224215544](https://github.com/milvus-io/milvus/assets/21117028/01af9a3a-59d9-4939-ab50-972b20df5a1b)

**Comment by yhmo** [comment]
看起来数据里有重复的id。你的截图上1.x的结果里第一条和第二条的id就是重复的。
对于重复id的处理，1.x和2.x行为不一样。1.x是把所有重复的都计算了返回给你，2.x是只计算时间戳最近的那个。
假设1690427114345443872这个id有3个重复的条目，第一个distance是0.501，第二个distance是1.109，第三个distance是1.800。那么，在2.x里，它只会认为distance是1.800，所以就排不进top10里。

**Reply by zxmin205** [reply]
哦，好的，确实是之前在1.x里对所有数据重建过，我清下数据再在2.x里试试，谢谢了

**Reply by zxmin205** [reply]
你说的是对的，确实在1.x里总的有3条相同的ID，原因知道了就好处理了，非常感谢！![image](https://github.com/milvus-io/milvus/assets/21117028/3baadd26-ed4a-44a3-a1b6-04cceb7ae486)
